### PR TITLE
SAK-42232 Calendar: Fix an issue with the iCal 3.0.7 upgrade

### PR DESCRIPTION
**Pull request by mpellicer**

### CODE REVIEW

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java [hunk 1]
```diff
@@ -65,6 +64,7 @@
 	public IcalendarReader()
 	{
 		super();
+		System.setProperty("net.fortuna.ical4j.timezone.cache.impl", MapTimeZoneCache.class.getName());
```
Comment (adrianfish) [inline calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java]: does this need to be set back to what it was before? In other words, might this affect other parts of Sakai, or is that just not a concern?

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java [hunk 2]
```diff
@@ -135,9 +135,9 @@ public String importStreamFromDelimitedFile(
 				for (Iterator j = list.iterator(); j.hasNext();) 
 				{
 					Period period = (Period) j.next();
-					TemporalAmount duration = period.getDuration();
-					long durationminutes = duration.get(ChronoUnit.MINUTES);
-					long durationhours = duration.get(ChronoUnit.HOURS);
+					Duration duration = Duration.from(period.getDuration());
+					long durationminutes = duration.toMinutes();
+					long durationhours =  duration.toHours();
 					//todo investiage ical4j's handling of 'days'
```
[the remark this answers is not in the document]
Comment (mpellicer) [reply calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java]: Don't think so, removing it.

File: calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java (hunk 1 above)
Comment (mpellicer) [reply calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java]: This property is needed in the newer version of the library, you can specify an "ical4j.properties" file or set the value at system level, I don't think this is dangerous for other parts of Sakai.

Comment (ottenhoff) [reply calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/readers/IcalendarReader.java]: Agree. This looks good to me.

**Pull request merged**